docs/examples/spiking-fashion-mnist.ipynb

```diff
@@ -157,17 +157,11 @@
     "        # add temporal dimension to the input shape; we can set it to None,\n",
     "        # to allow the model to flexibly run for different lengths of time\n",
     "        tf.keras.layers.Reshape((-1, 28 * 28), input_shape=(None, 28, 28)),\n",
-    "        # we can use Keras' TimeDistributed wrapper to allow the Dense layer\n",
-    "        # to operate on temporal data\n",
-    "        tf.keras.layers.TimeDistributed(tf.keras.layers.Dense(128)),\n",
+    "        # apply the Dense layer (without \"relu\")\n",
+    "        tf.keras.layers.Dense(128),\n",
     "        # replace the \"relu\" activation in the non-spiking model with a\n",
-    "        # spiking equivalent.\n",
-    "        # we'll learn more about \"spiking aware training\" later on\n",
+    "        # spiking equivalent\n",
     "        keras_spiking.SpikingActivation(\"relu\", spiking_aware_training=False),\n",
-    "        # we don't need TimeDistributed on this layer, because by default\n",
-    "        # SpikingActivation (like all Keras RNNs) only returns the values\n",
-    "        # from the last timestep. we could set return_sequences=True in\n",
-    "        # SpikingActivation if we wanted data for all timesteps.\n",
```

Comment:
Was this correct, and we're just not going into this detail here anymore? My impression was that `SpikingActivation` returned sequences by default, so that if you've got a network with a number of `SpikingActivation` layers, they're all returning the full spike timeseries so that subsequent layers can process the full spiketrain. Looking at the code, though, it looks like `return_sequences` defaults to `False`,  which would mean if you have multiple `SpikingActivation` layers in a network, the first one would collapse all of time (i.e. just return the last timestep), so subsequent layers would be computing spikes on one timestep (which may or may not be what you want, depending on whether you're using a dithering-type approach with multiple spikes per timestep, or a more traditional spiking approach where spikes are much sparser in time). My inclination is that most users will be using a more traditional spiking approach (at least to start), so it makes more sense to have `return_sequences=True` by default. Also, that makes the code easier to understand IMO, since `SpikingActivation` will return a tensor of the same shape unless it's got `return_sequences=False`.

Reply:
The `return_sequences` behaviour is unchanged, you just don't need a `TimeDistributed` layer in either case so I took out the comment.

The main motivation for having the default be `return_sequences=False` is that that is the default behaviour for Keras RNNs (which this whole API is heavily based on). So my thinking was that if users are used to how Keras RNNs work, and they make the connection that SpikingActivation layers are a lot like RNNs (which we're explicitly encouraging them to make), it'd be surprising if that one parameter had a different default. So either way has a potential to surprise users I guess. I'd be fine changing the default to `return_sequences=True` though, since I don't have a strong sense of which option is going to be more or less surprising to people.

Reply:
Looking at the example, it's actually got `return_sequences=False` for the training, and `return_sequences=True` for the testing. The reason for this is to "see" the spikes during testing (the spiking network is still using only the last timestep for classificiation), but I wonder if this might actually be a pretty common paradigm. When I've trained spiking networks before, I've always just done the training using rate neurons, so then I only need one timestep. During testing, though, I use many timesteps throughout the network. However, the way I'd accomplish that with this package is just to use input data with one timestep for training and many timesteps for testing, keeping `return_sequences=True` the whole time. Even with spiking-aware training, I could see a common paradigm being to train for one timestep with `dt` equal to the full simulation length you want during testing, and then testing with a much smaller `dt` (i.e. `dt_test = dt_train / n_steps`).

I think it would actually be more common to want `return_sequences=False` on filtering layers, since then you're not just taking the last timestep of a spiking layer (which may or may not be a spike), but rather the last timestep of a filter, which incorporates information over many layers.

I'm going to make `return_sequences=True` the default. It's possible that that will be surprising to some people who are familiar with the Keras RNNs, but even those people will not necessarily be thinking of the `SpikingActivation` layer as an RNN unless they've gone and looked at the code.